src/mol-io/reader/lammps/data/parser.ts

```diff
@@ -0,0 +1,212 @@
+import { ReaderResult as Result } from '../../result';
+import { TokenColumnProvider as TokenColumn } from '../../common/text/column/token';
+import { Column } from '../../../../mol-data/db';
+import { LammpsDataFile } from '../schema';
+
+const { readLine, skipWhitespace, eatValue, eatLine, markStart } = Tokenizer;
+
+const reWhitespace = /\s+/;
+
+function State(tokenizer: Tokenizer, runtimeCtx: RuntimeContext) {
+    return {
+        tokenizer,
+        runtimeCtx,
+    };
+}
+type State = ReturnType<typeof State>
+
+async function handleAtoms(state: State, count: number, atom_style: string): Promise<LammpsDataFile['atoms']> {
+    const { tokenizer } = state;
+    // default atom style is atomic
+    // depending on the atom style the number of columns can change
+    // Dictionary of atom_style and the number of columns
+    const atomStyleColumnMap: Record<string, number> = {
+        full: 7,
+        atomic: 5,
+        bond: 6,
+    };
+    const n = atomStyleColumnMap[atom_style];
+    const atomId = TokenBuilder.create(tokenizer.data, count * 2);
+    const moleculeType = TokenBuilder.create(tokenizer.data, count * 2);
+    const atomType = TokenBuilder.create(tokenizer.data, count * 2);
+    const charge = TokenBuilder.create(tokenizer.data, count * 2);
+    const x = TokenBuilder.create(tokenizer.data, count * 2);
+    const y = TokenBuilder.create(tokenizer.data, count * 2);
+    const z = TokenBuilder.create(tokenizer.data, count * 2);
+
+    const { position } = tokenizer;
+    readLine(tokenizer).trim();
+    tokenizer.position = position;
+
+    const { length } = tokenizer;
+    let linesAlreadyRead = 0;
+
+    await chunkedSubtask(state.runtimeCtx, 100000, void 0, chunkSize => {
+        const linesToRead = Math.min(count - linesAlreadyRead, chunkSize);
+        for (let i = 0; i < linesToRead; ++i) {
+            for (let j = 0; j < n; ++j) {
+                skipWhitespace(tokenizer);
+                markStart(tokenizer);
+                eatValue(tokenizer);
+                switch (atom_style) {
+                    case 'full': {
+                        switch (j) {
+                            case 0: TokenBuilder.addUnchecked(atomId, tokenizer.tokenStart, tokenizer.tokenEnd); break;
+                            case 1: TokenBuilder.addUnchecked(moleculeType, tokenizer.tokenStart, tokenizer.tokenEnd); break;
+                            case 2: TokenBuilder.addUnchecked(atomType, tokenizer.tokenStart, tokenizer.tokenEnd); break;
+                            case 3: TokenBuilder.addUnchecked(charge, tokenizer.tokenStart, tokenizer.tokenEnd); break;
+                            case 4: TokenBuilder.addUnchecked(x, tokenizer.tokenStart, tokenizer.tokenEnd); break;
+                            case 5: TokenBuilder.addUnchecked(y, tokenizer.tokenStart, tokenizer.tokenEnd); break;
+                            case 6: TokenBuilder.addUnchecked(z, tokenizer.tokenStart, tokenizer.tokenEnd); break;
```

Comment:
Instead of this, I would recommend doing:

```ts
const columns = {
  full: [atomId, moleculeType, ...],
  atomic: [atomId, atomType, ...],
  // ...
}

// ...

const column = columns[atom_style]?.[j];
if (column)  TokenBuilder.addUnchecked(column, tokenizer.tokenStart, tokenizer.tokenEnd);
```